platform/wayland/cog-platform-wl.c

```diff
@@ -105,90 +104,36 @@ noop()
 {
 }
 
-static gboolean
-wl_src_prepare(GSource *base, gint *timeout)
-{
-    struct wl_event_source *src = (struct wl_event_source *) base;
-
-    *timeout = -1;
-
-    while (wl_display_prepare_read(src->display) != 0) {
-        if (wl_display_dispatch_pending(src->display) < 0)
-            return false;
-    }
-    wl_display_flush(src->display);
-
-    return false;
-}
-
-static gboolean
-wl_src_check(GSource *base)
-{
-    struct wl_event_source *src = (struct wl_event_source *) base;
-
-    if (src->pfd.revents & G_IO_IN) {
-        if (wl_display_read_events(src->display) < 0)
-            return false;
-        return true;
-    } else {
-        wl_display_cancel_read(src->display);
-        return false;
-    }
-}
-
-static gboolean
-wl_src_dispatch(GSource *base, GSourceFunc callback, gpointer user_data)
-{
-    struct wl_event_source *src = (struct wl_event_source *) base;
-
-    if (src->pfd.revents & G_IO_IN) {
-        if (wl_display_dispatch_pending(src->display) < 0)
-            return false;
-    }
-
-    if (src->pfd.revents & (G_IO_ERR | G_IO_HUP))
-        return false;
-
-    src->pfd.revents = 0;
-
-    return true;
-}
+static void display_popup(void);
+static void update_popup(void);
+static void destroy_popup(void);
 
-static void
-wl_src_finalize(GSource *base)
-{
-}
+static void seat_on_capabilities(void *, struct wl_seat *, uint32_t);
+static void seat_on_name(void *, struct wl_seat *, const char *);
 
-static GSource *
-setup_wayland_event_source(GMainContext *main_context, struct wl_display *display)
+/**
+ * CogWlPlatform::cog_wl_compositor_create_surface:
+ * @compositor: The Wayland compositor.
+ * @container: User data.
+ *
+ * Commonly used for adding reference to the container that has this surface.
+ * For example, for a surface on a CogWlWindow, the reference will point to
+ * the CogWlPlatform. The rationale behind this is to articulate an easy way
+ * for passing the container as contextual information associated to a
+ * wl_surface. This is interesting for avoiding the use of global
+ * references and, in a near future, will make easier the access to the
+ * object that groups the CogWlViews.
+ *
+ * returns: static struct wl_surface *
+ */
+static struct wl_surface *
+cog_wl_compositor_create_surface(struct wl_compositor *compositor, void *container)
```

Comment:
Now that the patch uses `cog_platform_get_default()`, we do not need this anymore. We can remove this helper function and leave the call to `wl_compositor_create_surface()` where it was before.

Reply:
Good suggestion. Applied in the ultimate revision (https://github.com/Igalia/cog/compare/a43b0fc0cdf500b5ef97210b5129718c00f0186f..5896b0bb752608ea4d4ef70b4102a0cdb4e69381).